one-ui/src/test/java/com/marklogic/hub/oneui/TestHelper.java

```diff
@@ -120,24 +127,48 @@ public void addStagingDoc(String uri, DocumentMetadataHandle meta, String resour
     public File getResourceFile(String resourceName) {
         return new File(Objects.requireNonNull(TestHelper.class.getClassLoader().getResource(resourceName)).getFile());
     }
-    private void createUser(String username, String password, String role) {
+
+    public ArtifactManagerImpl getArtifactManager() {
+        return new ArtifactManagerImpl(hubConfig);
+    }
+
+    public String getLoginPayload(String role)
+        throws JsonProcessingException {
+        LoginInfo loginInfo = new LoginInfo();
+        loginInfo.password = "password";
+        loginInfo.mlHost = mlHost;
+        switch (role) {
+            case "data-hub-developer":
+                loginInfo.username = dataHubDeveloperUsername;;
+                break;
+            case "data-hub-environment-manager":
+                loginInfo.username = dataHubEnvironmentManagerUsername;;
+                break;
+            case "admin":
+                loginInfo.username = adminUserName;;
+                break;
+            default:
+                loginInfo.username = "fake";
+                break;
+        }
+        return new ObjectMapper().writeValueAsString(loginInfo);
+    }
+
+    /*  EnvironmentControllerTest.testUpload* uninstalls/installs datahub, hence the roles have to be reassigned to users
+    after datahub is reinstalled. */
+    public void assignRoleToUsers() {
```

Comment:
You'll also need to apply the database properties that Installer applies. 

I think the better approach is that after DHF is re-installed (and I assume it's being installed in the same fashion that "gradle bootstrap" installs it?), call a new static method on Installer that creates the additional users and applies the database properties. Call that method something like "afterInstall", and it'll have params of a DataHub and a HubConfig. That ensures that Installer does all of its additional post-install work.

Reply:
I will be calling Installer.afterInstall() from tests in one-ui proj.  If I import Installer to the tests in one-ui,
1. HubtestBase.afterPropertiesSet() gets called which attempts to populate objects in HubTestBase. In order to do that, I will have to copy gradle.properties from marklogic-data-hub project to one-ui project (similar to what we have for web project). Is it ok to do that ?
2. I should be able to access test resources of marklogic-data-hub project  in order to update the database properties. Do you know of a way test resource from marklogic-data-hub project be used in one-ui rather than making an explicit copy ? I couldn't find the

Reply:
The installation tests have been moved to a different test suite